src/main/webapp/app/text-assessment/text-assessment.component.ts

```diff
@@ -0,0 +1,132 @@
+    text: string;
+    participation: Participation;
+    submission: TextSubmission;
+    result: Result;
+    assessments: Feedback[] = [];
+    exercise: TextExercise;
+    totalScore = 0;
+    assessmentsAreValid: boolean;
+    invalidError: string;
+    isAuthorized = true;
+    accountId = 0;
+    busy = true;
+
+    public getColorForIndex = HighlightColors.forIndex;
+
+    constructor(
+        private jhiAlertService: JhiAlertService,
+        private modalService: NgbModal,
+        private router: Router,
+        private route: ActivatedRoute,
+        private resultService: ResultService,
+        private assessmentsService: TextAssessmentsService
+    ) {
+        this.assessments = [];
+        this.assessmentsAreValid = false;
+    }
+
+    public ngOnInit(): void {
+        this.busy = true;
+        const exerciseId = Number(this.route.snapshot.paramMap.get('exerciseId'));
+        const submissionId = Number(this.route.snapshot.paramMap.get('submissionId'));
+
+        this.assessmentsService.getFeedbackDataForExerciseSubmission(exerciseId, submissionId).subscribe(participation => {
+            this.participation = participation;
+            this.submission = <TextSubmission>this.participation.submissions[0];
+            this.exercise = <TextExercise>this.participation.exercise;
+            this.result = this.participation.results[0];
+            this.assessments = this.result.feedbacks;
+            if (!this.assessments) {
+                this.assessments = [];
+            }
+            this.busy = false;
+        });
+    }
+
+    public addAssessment(assessmentText: string): void {
+        const assessment = new Feedback();
+        assessment.reference = assessmentText;
+        assessment.credits = 0;
+        this.assessments.push(assessment);
+        this.checkScoreBoundaries();
+    }
+
+    public deleteAssessment(assessmentToDelete: Feedback): void {
+        this.assessments = this.assessments.filter(elem => elem !== assessmentToDelete);
+        this.checkScoreBoundaries();
+    }
+
+    public save(): void {
+        this.checkScoreBoundaries();
+        if (!this.assessmentsAreValid) {
+            this.jhiAlertService.error('arTeMiSApp.textAssessment.invalidAssessments');
+            return;
+        }
+
+        this.assessmentsService.save(this.assessments, this.exercise.id, this.result.id).subscribe(response => {
+            this.result = response.body;
+            this.jhiAlertService.success('arTeMiSApp.textAssessment.saveSuccessful');
+        });
+    }
+
+    public submit(): void {
+        this.checkScoreBoundaries();
+        if (!this.assessmentsAreValid) {
+            this.jhiAlertService.error('arTeMiSApp.textAssessment.invalidAssessments');
+            return;
+        }
+
+        this.assessmentsService.submit(this.assessments, this.exercise.id, this.result.id).subscribe(response => {
+            response.body.participation.results = [response.body];
```

Comment:
Why are you creating a circular reference inside the object?

Reply:
👍